tests/qchem/test_tapering.py

```diff
@@ -842,7 +853,7 @@ def test_consistent_taper_ops(operation, op_gen):
             hamiltonian,  # for energy
             sum(qml.PauliZ(wire) for wire in hamiltonian.wires),  # for local-cost 1
             # for local-cost 2
-            sum(qml.PauliZ(wire) @ qml.PauliZ(wire + 1) for wire in hamiltonian.wires),
+            sum(qml.PauliZ(wire) @ qml.PauliZ(wire + 1) for wire in hamiltonian.wires[:-1]),
```

Comment:
Otherwise, `wire+1` would lead to having operations that act on wires not provided via `wire_order`.

Reply:
Why it was not an issue before this PR?

Reply:
I guess this behavior comes from `wire_order` not needing all the wires in the case of tensors. A minimal example could be this - 
![image](https://github.com/PennyLaneAI/pennylane/assets/23234279/14a123a2-5c3c-488c-a054-daec78d2de27)